cmd/kubeadm/app/constants/constants.go

```diff
@@ -446,36 +442,25 @@ func AddSelfHostedPrefix(componentName string) string {
 	return fmt.Sprintf("%s%s", SelfHostingPrefix, componentName)
 }
 
-// CreateTempDirForKubeadm is a function that creates a temporary directory under /etc/kubernetes/tmp (not using /tmp as that would potentially be dangerous)
-func CreateTempDirForKubeadm(dirName string) (string, error) {
-	tempDir := path.Join(KubernetesDir, TempDirForKubeadm)
-	// creates target folder if not already exists
-	if err := os.MkdirAll(tempDir, 0700); err != nil {
-		return "", errors.Wrapf(err, "failed to create directory %q", tempDir)
+// CreateTempDirForKubeadm is a function that creates a temporary directory with name dirName in path prefix.
+// If prefix is empty, /etc/kubernetes/tmp directory will be used.
+func CreateTempDirForKubeadm(prefix, dirName string) (string, error) {
+	if prefix == "" {
+		prefix = filepath.Join(KubernetesDir, "tmp")
+		if err := os.MkdirAll(prefix, 0700); err != nil {
+			return "", errors.Wrap(err, "couldn't create prefix directory")
```

Comment:
added some minor comments
```suggestion
			return "", errors.Wrap(err, "failed to create directory %q", prefix)
```

Reply:
This suggestion will need a `Wrapf`, I'll update :)